src/simple-keyring.ts

```diff
@@ -0,0 +1,226 @@
+    });
+  }
+
+  async serialize() {
+    return this.#wallets.map((a) => a.privateKey.toString('hex'));
+  }
+
+  async deserialize(privateKeys: string[]) {
+    this.#wallets = privateKeys.map((hexPrivateKey) => {
+      const strippedHexPrivateKey = stripHexPrefix(hexPrivateKey);
+      const privateKey = Buffer.from(strippedHexPrivateKey, 'hex');
+      const publicKey = privateToPublic(privateKey);
+      return { privateKey, publicKey };
+    });
+  }
+
+  async addAccounts(numAccounts = 1) {
+    const newWallets = [];
+    for (let i = 0; i < numAccounts; i++) {
+      const privateKey = this.#generateKey();
+      const publicKey = privateToPublic(privateKey);
+      newWallets.push({ privateKey, publicKey });
+    }
+    this.#wallets = this.#wallets.concat(newWallets);
+    const hexWallets = newWallets.map(({ publicKey }) =>
+      add0x(bufferToHex(publicToAddress(publicKey))),
+    );
+    return hexWallets;
+  }
+
+  async getAccounts() {
+    return this.#wallets.map(({ publicKey }) =>
+      add0x(bufferToHex(publicToAddress(publicKey))),
+    );
+  }
+
+  async signTransaction(
+    address: Hex,
+    transaction: TypedTransaction,
+    options: Record<string, unknown> = {},
+  ): Promise<TxData> {
+    const privKey = this.#getPrivateKeyFor(address, options);
+    const signedTx = transaction.sign(privKey);
+    // Newer versions of Ethereumjs-tx are immutable and return a new tx object
+    return signedTx === undefined ? transaction : signedTx;
+  }
+
+  // For eth_sign, we need to sign arbitrary data:
+  async signMessage(
+    address: Hex,
+    data: string,
+    opts = { withAppKeyOrigin: '' },
+  ) {
+    const message = stripHexPrefix(data);
+    const privKey = this.#getPrivateKeyFor(address, opts);
+    const msgSig = ecsign(Buffer.from(message, 'hex'), privKey);
+    const rawMsgSig = concatSig(toBuffer(msgSig.v), msgSig.r, msgSig.s);
+    return rawMsgSig;
+  }
+
+  // For personal_sign, we need to prefix the message:
+  async signPersonalMessage(
+    address: Hex,
+    msgHex: Hex,
+    opts = { withAppKeyOrigin: '' },
+  ) {
+    const privKey = this.#getPrivateKeyFor(address, opts);
+    // const privateKey = Buffer.from(privKey, 'hex');
+    return personalSign({ privateKey: privKey, data: msgHex });
+  }
+
+  // For eth_decryptMessage:
+  async decryptMessage(withAccount: Hex, encryptedData: Eip1024EncryptedData) {
+    const wallet = this.#getWalletForAccount(withAccount);
+    const privateKey = wallet.privateKey.toString('hex');
+    return decrypt({ privateKey, encryptedData });
+  }
+
+  // personal_signTypedData, signs data along with the schema
+  async signTypedData(
+    address: Hex,
+    typedData: any,
+    opts: Record<string, string> = { version: SignTypedDataVersion.V1 },
+  ) {
+    // Treat invalid versions as "V1"
+    const version = Object.keys(SignTypedDataVersion).includes(
+      opts.version || '',
+    )
+      ? (opts.version as SignTypedDataVersion)
```

Comment:
Is there any way to avoid this typecast? This would allow `''` to be typecast as `SignTypedDataVersion`, which doesn't seem like something we'd want.

Reply:
This is tricky, but I'm working on it.  The problem is that we want to accept any string, and if it's not in the enum, use `SignTypedDataVersion.V1`